docs/rfcs/rfc-021-global-authentication.md

```diff
@@ -0,0 +1,66 @@
+# Global authentication
+
+## Problem
+
+For some use cases the GraphQL API needs to be secured globally to restrict access to _any_ of the top-level types without prior authentication. This is also referred to as [API-wide authorization](https://www.apollographql.com/docs/apollo-server/security/authentication/#api-wide-authorization).
+Currently to achieve this every type or union needs to be extended with an `@auth` directive: `@auth(rules: [{ isAuthenticated: true }])`.
+
+## Proposed Solution
+
+Add a configuration option, a boolean, to `Neo4jGraphQLConfig` which allows for an opt-in for global authentication. The default value is `false`.
+
+```javascript
+const neoSchema = new Neo4jGraphQL({
+    config: {
+        enableDebug: false,
+        globalAuthentication: true,
+    },
+    typeDefs: typeDefs,
+    driver,
+    plugins: {
+        auth: new Neo4jGraphQLAuthJWTPlugin({
+            secret: "1234",
+            noVerify: false,
+        }),
+    },
```

Comment:
Just a quick parting comment on this before I head off on vacation - I believe that the configuration for global authentication should be in the auth plugin, and not in the library. It depends on JWT decoding and verification, which is a plugin feature.

Reply:
Really good point @darrellwarde! Will make the necessary adjustments